modules/oe_theme_inpage_navigation/tests/src/FunctionalJavascript/InPageNavigationBlockTest.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_theme_inpage_navigation\FunctionalJavascript;
+
+use Drupal\FunctionalJavascriptTests\WebDriverTestBase;
+use Drupal\media\MediaInterface;
+use Drupal\node\Entity\NodeType;
+use Drupal\Tests\oe_theme\PatternAssertions\InPageNavigationAssert;
+
+/**
+ * Test Inpage navigation block plugin.
+ */
+class InPageNavigationBlockTest extends WebDriverTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'block',
+    'oe_theme_helper',
+    'oe_theme_content_publication',
+    'oe_theme_content_organisation',
+    'oe_theme_content_organisation_reference',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    parent::setUp();
+
+    // Enable and set OpenEuropa Theme as default.
+    $this->container->get('theme_installer')->install(['oe_theme']);
+    $this->container->get('theme_handler')->setDefault('oe_theme');
+    $this->container->set('theme.registry', NULL);
+
+    // Enable test module with placed block in new region and overriden
+    // legacy inpage navigation template.
+    $this->container->get('module_installer')
+      ->install(['oe_theme_inpage_navigation_test']);
+
+    // Rebuild the ui_pattern definitions to collect the ones provided by
+    // oe_theme itself.
+    \Drupal::service('plugin.manager.ui_patterns')->clearCachedDefinitions();
+
+    \Drupal::service('emr.installer')->installEntityMetaTypeOnContentEntityType('oe_theme_inpage_navigation', 'node', ['oe_publication']);
+
+    // Enable inpage_navigation for Publication content.
+    $ct_with = NodeType::load('oe_publication');
+    $ct_with->setThirdPartySetting('oe_theme_inpage_navigation', 'enabled', TRUE);
+    $ct_with->save();
+  }
+
+  /**
+   * Test in-page entity meta relation plugin.
+   */
+  public function testInPageNavigationForm(): void {
+
+    // Create a document for Publication.
+    $media_document = $this->createMediaDocument('publication_document');
+
+    // Create a Publication node with required fields only.
```

Comment:
Let's test also a content type without inpage navigation elements like page and test that h2 added in content regions are correctly handled.

Reply:
Added additional test.